sync/http/middleware.go

```diff
@@ -99,14 +100,33 @@ func NewAuthMiddleware(auth auth.Authenticator) MiddlewareFunc {
 	}
 }
 
-// NewTracingMiddleware creates a MiddlewareFunc that continues a tracing span and finishes it.
-func NewTracingMiddleware(path string) MiddlewareFunc {
+// NewLoggingTracingMiddleware creates a MiddlewareFunc that continues a tracing span and finishes it.
+// It also logs the HTTP request on debug logging level
+func NewLoggingTracingMiddleware(path string) MiddlewareFunc {
 	return func(next http.Handler) http.Handler {
 		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 			sp, r := trace.HTTPSpan(path, r)
 			lw := newResponseWriter(w)
 			next.ServeHTTP(lw, r)
 			trace.FinishHTTPSpan(sp, lw.Status())
+			if log.Enabled(log.DebugLevel) {
+				remoteAddr := r.RemoteAddr
+				if i := strings.LastIndex(remoteAddr, ":"); i != -1 {
+					remoteAddr = remoteAddr[:i]
+				}
+				info := map[string]interface{}{
+					"request": map[string]interface{}{
+						"remote-address": remoteAddr,
+						"method":         r.Method,
+						"url":            r.URL,
+						"proto":          r.Proto,
+						"status":         lw.status,
```

Comment:
```suggestion
						"status":         lw.Status(),
```
just to be as before